src/test/java/me/itzg/helpers/users/ManageUsersCommandTest.java

```diff
@@ -445,6 +445,215 @@ void givenNames(WireMockRuntimeInfo wmInfo) {
+                        )
+                );
+        }
+        
+        @Test
+        void partialOnlineOffline(WireMockRuntimeInfo wmInfo) {
+            setupUserStubs();
+
+            final int exitCode = new CommandLine(
+                new ManageUsersCommand()
+            )
+                .execute(
+                    "--mojang-api-base-url", wmInfo.getHttpBaseUrl(),
+                    "--user-api-provider", "mojang",
+                    "--offline",
+                    "--type", "JAVA_WHITELIST",
+                    "--output-directory", tempDir.toString(),
+                    "user1:offline", "user2"
+                );
+
+            assertThat(exitCode).isEqualTo(0);
+
+            final Path expectedFile = tempDir.resolve("whitelist.json");
+
+            assertThat(expectedFile).exists();
+
+            assertJson(expectedFile)
+                .isArrayContainingExactlyInAnyOrder(
+                    conditions()
+                        .satisfies(conditions()
+                            .at("/name").hasValue("user1")
+                            .at("/uuid").hasValue(USER1_OFFLINE_UUID)
+                        )
+                        .satisfies(conditions()
+                            .at("/name").hasValue("user2")
+                            .at("/uuid").hasValue(USER2_UUID)
+                        )
+                );
+        }
+    
+        @Test
+        void givenNameExistsInCache(WireMockRuntimeInfo wmInfo) throws IOException {
+            setupUserStubs();
+
+            final Path expectedFile = tempDir.resolve("usercache.json");
+            Files.write(expectedFile,
+                Collections.singletonList(
+                    "[{\"name\":\"user1\",\"uuid\":\"" + USER1_UUID
+                        + "\",\"expiresOn\":\"2023-09-13 20:14:26 +0000\"}]"
+                )
+            );
+
+            final int exitCode = new CommandLine(
+                new ManageUsersCommand()
+            )
+                .execute(
+                    "--mojang-api-base-url", wmInfo.getHttpBaseUrl(),
+                    "--user-api-provider", "mojang",
+                    "--type", "JAVA_WHITELIST",
+                    "--output-directory", tempDir.toString(),
+                    "--offline",
+                    "user1:offline"
+                );
+
+            assertThat(exitCode).isEqualTo(0);
+
+            final Path whitelistFile = tempDir.resolve("whitelist.json");
+            assertThat(whitelistFile).exists();
+
+            assertJson(whitelistFile)
+                .isArrayContainingExactlyInAnyOrder(
+                    conditions()
+                        .satisfies(conditions()
+                            .at("/name").hasValue("user1")
+                            .at("/uuid").hasValue(USER1_OFFLINE_UUID)
+                        )
+                );
+
+            verify(0, getRequestedFor(urlEqualTo("/users/profiles/minecraft/user1")));
```

Comment:
Ah, glad to see you have this case to balance the assertion I commented previously. Thanks.